vars/maven.groovy

```diff
@@ -27,3 +27,17 @@ def runMavenWithSubmarineSettings(String goals, boolean skipTests, String logFil
 def runMavenWithSubmarineSettings(String goals, Properties properties, String logFileName = null) {
     runMavenWithSettings('9239af2e-46e3-4ba3-8dd6-1a814fc8a56d', goals, properties, logFileName)
 }
+/**
+ *
+ * @param settingsXmlId settings.xml file
+ * @param goals maven gals
+ * @param sonarCloudId Jenkins token for SonarCloud*
+ */
+def runMavenWithSettingsSonar(String settingsXmlId, String goals, String sonarCloudId) {
```

Comment:
I would replace this method by this (to log maven into a file)
```groovy
def runMavenWithSettingsSonar(String settingsXmlId, String goals, String sonarCloudId, String logFileName = null) {
    configFileProvider([configFile(fileId: settingsXmlId, variable: 'MAVEN_SETTINGS_XML')]) {
        withCredentials([string(credentialsId: sonarCloudId, variable: 'TOKEN')]) {
            def teeCommand = logFileName ? '| tee $WORKSPACE/'+ logFileName + ' ; test ${PIPESTATUS[0]} -eq 0' : ''
            sh "mvn -B -s $MAVEN_SETTINGS_XML -Dsonar.login=${TOKEN} ${goals}  ${teeCommand}"
        }
    }
}
```
The maven.log artifacts will be stored by this https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/dc97bcfaebe72cd52d2a20fffd6b8e554f80dc07/Jenkinsfile#L60

Reply:
done